test/algorithms/test_qft.cpp

```diff
@@ -176,34 +181,47 @@ TEST_P(QFT, Simulation) {
                 dd::RealNumber::eps);
     EXPECT_NEAR(c.imag(), 0, dd::RealNumber::eps);
   }
+  dd->decRef(sim);
 }
 
 TEST_P(QFT, FunctionalityRecursiveEquality) {
   // there should be no error constructing the circuit
   ASSERT_NO_THROW({ qc = std::make_unique<qc::QFT>(nqubits, false); });
+  auto exp = std::make_unique<dd::FunctionalityConstructionExperiment>();
+  exp->dd = std::make_unique<dd::Package<>>(nqubits);
+  initialCacheCount = exp->dd->cn.cacheCount();
+  initialComplexCount = exp->dd->cn.realCount();
 
   // there should be no error building the functionality recursively
-  ASSERT_NO_THROW({ func = buildFunctionalityRecursive(qc.get(), dd); });
+  exp = dd::benchmarkFunctionalityConstruction(*qc);
+  auto func = exp->func;
 
   // there should be no error building the functionality regularly
-  qc::MatrixDD funcRec{};
-  ASSERT_NO_THROW({ funcRec = buildFunctionality(qc.get(), dd); });
+  auto funcRec = buildFunctionalityRecursive(qc.get(), exp->dd);
 
+  dd = std::move(exp->dd);
```

Comment:
move that statement up a bit and you can reduce even more of the changes from above.

Reply:
Just look at the changelog on GitHub ;o) there aren't that many lines to choose from.